File: backend/app/common/schemas/enum_meta.py
"""跨模块枚举展示元数据（value + label + hint）。

各业务域 ``tenant/*/meta.py`` 维护文案，经 ``GET /{module}/meta`` 下发；
前端统一用 ``EnumOption`` + ``optionLabel()`` 渲染，避免前后端枚举漂移。
"""

from __future__ import annotations

from enum import Enum
from typing import Iterable

from pydantic import BaseModel


class EnumOption(BaseModel):
    """单条枚举选项；``implemented`` 仅钩子触发器等需标注接线状态的域使用。"""

    value: str
    label: str
    hint: str | None = None
    implemented: bool | None = None


def enum_options(
    enum_cls: type[Enum],
    labels: dict[str, tuple[str, str | None]],
    *,
    implemented: dict[str, bool] | None = None,
) -> list[EnumOption]:
    """从枚举与标签表生成选项列表（顺序与枚举定义一致）。"""
    out: list[EnumOption] = []
    for member in enum_cls:
        key = member.value
        label, hint = labels.get(key, (key, None))
        out.append(
            EnumOption(
                value=key,
                label=label,
                hint=hint,
                implemented=implemented.get(key) if implemented else None,
            )
        )
    return out


def literal_options(pairs: Iterable[tuple[str, str, str | None]]) -> list[EnumOption]:
    """非枚举常量列表：[(value, label, hint?), ...]。"""
    return [EnumOption(value=v, label=lb, hint=h) for v, lb, h in pairs]

File: ui/workbench/lib/enum-meta.ts
/**
 * 全站枚举展示契约（与 backend `app/common/schemas/enum_meta.py` 一致）。
 *
 * 必要链路（文案单一来源，避免前后端枚举漂移）：
 *   1. `backend/app/tenant/<module>/meta.py` — 维护 value/label/hint
 *   2. `GET /<module>/meta` — 须在 `/{id}` 等路径参数路由之前注册
 *   3. `lib/api.ts` — `get*Meta()`
 *   4. `lib/enum-meta-cache.tsx` + `hooks/use-enum-meta.ts` + `hooks/use-*-meta.ts` — AppShell 内按 cacheKey 缓存 meta
 *   5. `lib/*-labels.ts` 或 `document-status.ts` — `optionLabel(meta?.field, value)` + 本地 fallback
 *   6. 页面/组件 — 将 `meta` 传入 label 函数或 `statusOptions` 等 props
 *
 * 纯 UI Tab（如「全部」「广场/安装」）可保留在前端，不必进 meta。
 * 模块对照表见 `docs/guides/hooks.md` §9；全站链路索引见 `lib/chains.ts` §4。
 */

export type EnumOption = {
  /** 与 API/ORM 存库值一致 */
  value: string;
  /** 列表、下拉、徽章展示文案 */
  label: string;
  hint?: string | null;
  /** 钩子：是否已接线实现 */
  implemented?: boolean | null;
};

/** 按 value 取 label；未命中时回退为 value 本身 */
export function optionLabel(options: EnumOption[] | undefined, value: string): string {
  return options?.find((o) => o.value === value)?.label ?? value;
}

/** 按 value 取 hint（表单说明、卡片副标题等） */
export function optionHint(options: EnumOption[] | undefined, value: string): string | undefined {
  const h = options?.find((o) => o.value === value)?.hint;
  return h ?? undefined;
}
